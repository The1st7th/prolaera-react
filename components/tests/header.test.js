import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import fs from 'fs';
import React from 'react';
import { renderEmail } from 'react-html-email';
import renderer from 'react-test-renderer';
import builderHeader, { Header } from '../header';

Enzyme.configure({ adapter: new Adapter() });

describe('Email Header Template', () => {
  //Create snapshot of header with custom url and text
  it('it checks snapshot with custom link url and text', async () => {
    const Header = await builderHeader('http://assets.prolaera.com/NHHCo-lg.png');
    let headerComponent = renderer.create(<Header text={'View My Compliance'} link={'https://www.google.com/'} />);
    let headerJson = headerComponent.toJSON();
    expect(headerJson).toMatchSnapshot();
  });

  // Use to save html to a file to make building easier
  it('it writes an html file with custom link url and text', async () => {
    const Header = await builderHeader('http://assets.prolaera.com/NHHCo-lg.png');
    const headerHtml = renderEmail(<Header text={'View My Compliance'} link={'https://www.google.com/'} />);
    const saved = await writeFile(headerHtml);
    expect(saved).toEqual(true);
  });

  //Use to test image resizing
  it('resizes image successfully NHHco Logo', async () => {
    const Header = await builderHeader('http://assets.prolaera.com/NHHCo-lg.png');
    const headerComponent = renderer.create(<Header />);
    let headerJson = headerComponent.toJSON();
    expect(headerJson).toMatchSnapshot();
  });
  it('resizes image successfully KRS logo', async () => {
    const Header = await builderHeader('http://assets.prolaera.com/KRS-lg.png');
    const headerComponent = renderer.create(<Header />);
    let headerJson = headerComponent.toJSON();
    expect(headerJson).toMatchSnapshot();
  });
  it('resizes image successfully a-lign logo', async () => {
    const Header = await builderHeader('http://assets.prolaera.com/a-lign-lg.png');
    const headerComponent = renderer.create(<Header />);
    let headerJson = headerComponent.toJSON();
    expect(headerJson).toMatchSnapshot();
  });
});

//Enzyme Tests for custom link url and link text
describe('<Header />', () => {
  it('checks that Header html contains logo', async () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('.headerLogo').exists()).toEqual(true);
  });
  it('checks that component html includes link text', async () => {
    const wrapper = shallow(<Header text={'View My Compliance'} />);
    expect(wrapper.contains(<a href="https://app.prolaera.com"> View My Compliance → </a>)).toBe(true);
  });
  it('checks that component html includes link url', async () => {
    const wrapper = shallow(<Header text={'View My Compliance'} link={'https://www.google.com/'} />);
    expect(wrapper.contains(<a href="https://www.google.com/"> View My Compliance → </a>)).toBe(true);
  });
});

async function writeFile(emailHtml) {
  return new Promise((resolve, reject) => {
    fs.writeFile(`${__dirname}/test.html`, emailHtml, err => {
      if (err) {
        console.log(err);
        reject(err);
      }
      resolve(true);
    });
  });
}
