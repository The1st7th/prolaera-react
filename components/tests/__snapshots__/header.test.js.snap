// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email with custom header link it checks snapshot with custom link url and text 1`] = `
<table
  align="center"
  border="0"
  cellPadding={0}
  cellSpacing={0}
  className="header"
  style={
    Object {
      "backgroundColor": "#F7F7F7",
    }
  }
  valign="top"
  width="100%"
>
  <tbody>
    <tr>
      <td
        align="center"
      >
        <table
          align="center"
          cellSpacing={0}
          style={
            Object {
              "maxWidth": "584px",
              "padding": "15px",
            }
          }
        >
          <tr
            align="left"
          >
            <td
              width="292px"
            >
              <img
                alt="logo"
                height={47}
                src="https://assets.prolaera.com/prolaeraLogo_fullText.png"
                style={
                  Object {
                    "border": "none",
                    "display": "block",
                    "outline": "none",
                    "textDecoration": "none",
                  }
                }
                width={200}
              />
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
              width="252px"
            >
              <a
                href="https://www.google.com/"
              >
                 
                View My Compliance
                 → 
              </a>
            </td>
          </tr>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Email with custom header link it renders and matches previous snapshot 1`] = `
<table
  align="center"
  border="0"
  cellPadding={0}
  cellSpacing={0}
  className="header"
  style={
    Object {
      "backgroundColor": "#F7F7F7",
    }
  }
  valign="top"
  width="100%"
>
  <tbody>
    <tr>
      <td
        align="center"
      >
        <table
          align="center"
          cellSpacing={0}
          style={
            Object {
              "maxWidth": "584px",
              "padding": "15px",
            }
          }
        >
          <tr
            align="left"
          >
            <td
              width="292px"
            >
              <img
                alt="logo"
                height={47}
                src="https://assets.prolaera.com/prolaeraLogo_fullText.png"
                style={
                  Object {
                    "border": "none",
                    "display": "block",
                    "outline": "none",
                    "textDecoration": "none",
                  }
                }
                width={200}
              />
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
              width="252px"
            >
              <a
                href="https://www.google.com/"
              >
                 
                View My Compliance
                 → 
              </a>
            </td>
          </tr>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;
